mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -4360,6 +4360,28 @@ impl OrderbookItem {
     }
 }
 
+trait AgeSec {
+    fn created_at(&self) -> u64;
+
+    fn age_sec(&self) -> Result<i64, String> {
+        let (now, created_at) = (now_sec(), self.created_at());
+        let err_fn = || format!("Failed to get age from (now, created at): ({now}, {created_at}");
+        now_sec()
+            .checked_sub(created_at)
+            .ok_or_else(err_fn)?
+            .to_i64()
+            .ok_or_else(err_fn)
+    }
+}
+
+impl AgeSec for OrderbookP2PItem {
+    fn created_at(&self) -> u64 { self.created_at }
+}
+
+impl AgeSec for OrderbookItem {
+    fn created_at(&self) -> u64 { self.created_at }
+}
+
```

Comment:
Are these multiple hops necessary for doing simple subtraction?

What about creating local macro on top of this module like:
```rs
macro_rules! try_sub {
    ($base: expr, $sub: expr) => {
        $base
            .checked_sub($sub)
            .ok_or_else(|| ERRL!("({} - {}) caused an integer overflow.", $base, $sub))?
            .try_into()
            .map_err(|e| ERRL!("{}", e))
    };
}
```

which then can be used like `let t: i64 = try_sub!(now_sec(), self.created_at)?;`

Reply:
done